install.sh

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+set -e
+
+version="2.2.0"
+
+# Do nothing if echidna is already installed
+if echidna-test --version > /dev/null 2>&1
+then
+  echo "WARNING: $(echidna-test --version) is installed but is out-of-date, we recommend removing it"
+fi
+
+# Do nothing if echidna is already installed
+if echidna --version > /dev/null 2>&1
+then
+  echo "$(echidna --version) is already installed" && exit
+  # TODO: check that the version is up to date, otherwise offer to upgrade
+fi
+
+# In the unlikely scenario that python3 isn't available, let the user install it manually
+if ! python3 --version > /dev/null 2>&1
+then echo "python3 is not available, please install it manually before attempting to install echidna" && exit 1
+fi
+
+# install pip3 if not already installed
+if ! pip3 --version > /dev/null 2>&1
+then
+  echo "pip3 is not available, installing it now.."
+  curl https://bootstrap.pypa.io/get-pip.py | python3
+fi
+
+# install slither if not already installed
+if ! slither --version > /dev/null 2>&1
+then
+  echo "slither is not available, installing it now.."
+  pip3 install slither-analyzer --user
+fi
+
+echo "Installing Echidna v$version.."
+
+BASE_DIR=${XDG_CONFIG_HOME:-$HOME}
+ECHIDNA_DIR=${ECHIDNA_DIR-"$BASE_DIR/.echidna"}
+ECHIDNA_BIN_DIR="$ECHIDNA_DIR/bin"
+ECHIDNA_ARCHIVE_DIR="$ECHIDNA_DIR/v$version"
+ECHIDNA_ARCHIVE_PATH="$ECHIDNA_ARCHIVE_DIR/echidna.tar.gz"
+ECHIDNA_DIGEST_PATH="$ECHIDNA_ARCHIVE_DIR/echidna.sha256"
+TRUE_BIN_PATH="$ECHIDNA_ARCHIVE_DIR/echidna/echidna"
+BIN_PATH="$ECHIDNA_BIN_DIR/echidna"
+
+if [[ "$(uname)" == "Darwin" ]]
+then arch="MacOS"
```

Comment:
we should differentiate ARM vs x86 builds, maybe by inspecting `uname -m`. Otherwise M1 users will get slow fuzzing performance.

Reply:
Is there a separate release for ARM vs x86 macs? Under release assets I only see MacOS vs Ubuntu vs Windows

Reply:
@bohendo  yeah, there's MacOS-aarch64 files, with a different structure inside the tarball, e.g see https://github.com/crytic/echidna/releases/tag/v2.1.0

These are manually built (with the nix bundle target in the readme), compressed and uploaded; it seems the latest release is missing them. We used to have a more comprehensive build process that generated a ready to upload archive and fixed TERMINFO on the resulting binaries (the current process has [this bug again](https://github.com/crytic/echidna/pull/731/files)), maybe we can bring that back to aid releases going forward?